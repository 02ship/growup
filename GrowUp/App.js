import React, {useState} from 'react';
import { View, Text, StyleSheet, ListView, FlatList } from 'react-native';
import Header from './components/Header';
import ListItem from './components/ListItem';
import { v4 as uuidv4 } from 'uuid';

const App = () => {
  const [items, setItems] = useState([
      {id: uuidv4(), text: 'Sweep'},
      {id: uuidv4(), text: 'Mop'},
      {id: uuidv4(), text: 'Dust'},
  ]);

  const deleteItem = (id) => {
    setItems(prevItems => {
      return prevItems.filter(item => item.id != id);
    });
  }

  return (
    <View style={styles.container}>
      <Header />
      <FlatList
        data={items}
        renderItem={({item}) => <ListItem item={item} deleteItem={deleteItem} />}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 60,
  },
});

export default App;
