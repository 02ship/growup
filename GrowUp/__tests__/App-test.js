/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import { shallow, mount } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {render, cleanup} from 'react-native-testing-library';

afterEach(cleanup);

it('renders correctly', () => {
  renderer.create(<App />);
});

it('should match snapshot', () => {
  const result = renderer.create(<App />).toJSON();
  expect(result).toMatchSnapshot();
});

// it('can add an item', () =>{
//   const rendered = mount(<App />);
//   console.log(rendered);
//   console.log(rendered.instance());
//   const task = rendered.instance().addItem("sweep");
//   expect();
// })
