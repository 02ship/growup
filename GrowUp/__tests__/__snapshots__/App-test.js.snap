// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 60,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "green",
        "height": 60,
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 23,
          "textAlign": "center",
        }
      }
    >
      GrowUp!
    </Text>
  </View>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Add Task..."
      rejectResponderTermination={true}
      style={
        Object {
          "fontSize": 16,
          "height": 60,
          "padding": 8,
        }
      }
      testID="test-placeholder"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "green",
          "margin": 5,
          "opacity": 1,
          "padding": 9,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ÔÅß
        </Text>
         Add Item
      </Text>
    </View>
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
